app/misc/repo.ts

```diff
@@ -258,11 +259,94 @@ function openRepository() {
     }
   }
 
+// works as a monitor for any change to the reference list
+function refreshList(verbose) {
+  Git.Repository.open(repoFullPath)
+    .then(function (repo) {
+      repo.getCurrentBranch()
+        .then(function () {
+          // grab the list
+          return repo.getReferences(Git.Reference.TYPE.LISTALL);
+        })
+        .then(function (refList) {
+          // sort refList alphabetically to get uniform order of the list
+          refList.sort();
+
+          // monitor any changes to reference list
+          if (lastRefList.length === refList.length && lastRefList.every(function(value, index) { return value.name() === refList[index].name()})) {
+            // no change to the ref list, do nothing
+            return;
+          }
+
+          // detects changes, refresh the lists
+          console.log("branch or tag changes detected... refreshing branch and tag list");
+          bname = {};
+          tags = {};
+          clearBranchAndTagElement();
+          for (let i = 0; i < refList.length; i++) {
+            if (verbose) { console.log("reference name: " + refList[i].name()); }
+            //get simplified name
```

Comment:
I would like to see this comment expanded, so we can understand how you are parsing the name out.